marconi-mamba/test/Spec/Marconi/Mamba/Api/Query/Indexers/Utxo.hs

```diff
@@ -72,3 +95,69 @@ queryTargetAddressTest = property $ do
   -- To remove once the assert above is fixed
   fmap Aeson.encode fetchedRows === fmap Aeson.encode fetchedRows
   fmap Aeson.encode rows === fmap Aeson.encode rows
+
+-- | Start a servant process and execute the client action
+--
+withUserApp
+  :: IndexerEnv
+  -> (Warp.Port -> IO (JsonRpcResponse String UtxoQueryResult)) -- ^ hoisted JSON-RPC client such that it can run in IO
+  -> IO (JsonRpcResponse String UtxoQueryResult)
+withUserApp env  = Warp.testWithApplication(pure $ marconiApp env)
+
+clientEnv :: String -> Warp.Port -> IO (JsonRpcResponse String UtxoQueryResult)
+clientEnv address port = do
+  manager' <- newManager defaultManagerSettings
+  let env = mkClientEnv manager' (baseUrl port)
+  let (rpcEcho :<|> rpcTargets :<|> rpcUtxos) = getClients env
+  -- test will fail if these calls/routes fail.
+  _ <- rpcEcho "client calling "
+  _ <- rpcTargets ""
+  rpcUtxos address
+
+
+jsonRpcTest :: Property
+jsonRpcTest =  property $ do
+  event <- forAll genEvent
+  env <- liftIO . UIQ.initializeEnv $ Nothing
+  let
+    callback :: Utxo.UtxoIndexer -> IO ()
+    callback = atomically . UIQ.writeTMVar' (env ^. uiIndexer) -- update the indexer
+  liftIO . mocUtxoWorker callback [event] $ Utxo.Depth 10
+  let (qAddresses :: [String]) =
+        Set.toList . Set.fromList . fmap (unpack . C.serialiseAddress)
+        . mapMaybe (addressAnyToShelley . Utxo._address)
+        . Set.toList . Utxo.ueUtxos
+        $ event
+  rpcResponses <- liftIO $ traverse (\a -> withUserApp env (clientEnv a) ) qAddresses
+  when (any hasFailed rpcResponses ) Hedgehog.failure
+  let inserted =  Set.fromList . Utxo.eventsToRows $ event
+  let fetched =  Set.fromList . concatMap getRows $ rpcResponses
+  length fetched  === length inserted
+  fetched === inserted
```

Comment:
@koslambrou  this test consistently fails after a few iteration, usually 5 - 8. 
The error has to do with incorrect conversion of ScriptData:
```
              ┃   │               , _datum =
              ┃   │ -                 Just (ScriptDataNumber (-17))
              ┃   │ +                 Just (ScriptDataNumber 0)
```